tests/robustness/engine/engine.go

```diff
@@ -277,3 +299,46 @@ func (e *Engine) init(ctx context.Context) error {
 
 	return e.Checker.VerifySnapshotMetadata()
 }
+
+// InitS3WithServer initializes the Engine with InitS3 for use with the server/client model.
+func (e *Engine) InitS3WithServer(ctx context.Context, bucketName, testRepoPath, metaRepoPath, addr string) error {
+	if err := e.MetaStore.ConnectOrCreateS3(bucketName, metaRepoPath); err != nil {
+		return err
+	}
+
+	cmd, err := e.TestRepo.ConnectOrCreateS3WithServer(addr, bucketName, testRepoPath)
+	if err != nil {
+		return err
+	}
+
+	e.serverCmd = cmd
+
+	return e.init(ctx)
+}
+
+// InitFilesystemWithServer initializes the Engine for testing the server/client model with a local filesystem repository.
+func (e *Engine) InitFilesystemWithServer(ctx context.Context, testRepoPath, metaRepoPath, addr string) error {
+	if err := e.MetaStore.ConnectOrCreateFilesystem(metaRepoPath); err != nil {
+		return err
+	}
+
+	cmd, err := e.TestRepo.ConnectOrCreateFilesystemWithServer(addr, testRepoPath)
+	if err != nil {
+		return err
+	}
+
+	e.serverCmd = cmd
+
+	return e.init(ctx)
+}
+
+// cleanUpServer cleans up the server process.
+func (e *Engine) cleanUpServer() {
+	if e.serverCmd != nil {
+		e.serverCmd.SysProcAttr = &syscall.SysProcAttr{
+			Pdeathsig: syscall.SIGTERM,
+		}
+
+		e.serverCmd.Run() //nolint:errcheck
```

Comment:
What does this do? What is the intent?

Reply:
Cleanup the server process created in background.

Reply:
But, what does it do? how does it work?

I may be wrong, but I don't think this works.

Reply:
`e.serverCmd` is the process that is the Kopia API Server, to kill this process, we need to send `SIGTERM` signal and run it. ref: https://medium.com/@ganeshmaharaj/clean-exit-of-golangs-exec-command-897832ac3fa5.
And this seems to work, as when I run this suite multiple times, I never get the `51515 port not available`, which use to happen when I didn't add this block of code.
I will check once more, and confirm you.

Reply:
What happens when `e.serverCmd.Run()` is called above?

We can address this in a separate PR.